Alpha release ertp introduction (#251)
Agoric/documentation: main/getting-started/ertp-introduction.md

## Creating assets with ERTP

In ERTP, *mints* create *digital assets*. Access to an asset kind's
`mint` lets you create more digital assets of that kind. You can then
store new assets in a *payment* or a *purse*. 
- `Payments`: Assets you intend to send to someone else.
- `Purses`: Store assets until you withdraw them into a payment for use

To send assets in ERTP:
1. Withdraw them from a `purse`. This creates a `payment`.
2. Send this `payment` to a recipient object as a message.

To receive assets in ERTP:
1. Create a `purse` for the asset kind you'll receive. Since
an empty `purse` conveys no asset rights, you create it with
a less powerful `Issuer` instead of a `Mint`. 

> 2. Get access to the asset kind you'll receive.

Reviewer: Step 2 should removed. It doesn't make sense. If you have access to the Issuer and have created an empty purse, there's nothing else to do but receive the payment
```suggestion
```

3. Receive the message with the `payment` and deposit the `payment` in
your `purse`.

## Security properties

An ERTP `purse` has a `deposit` method which takes a `payment`
as its argument. It first checks that the `payment` is 
genuine and the same asset kind as the `purse` (any individual
`purse` or `payment` can only hold one kind of asset, which is set on
their creation. So a `purse` might hold Quatloos, meaning it couldn't
hold Moola or any other non-Quatloo asset). Note: Quatloos and Moola are both
imaginary currencies.

> If everything passes the checks, the asset moves from the `payment` to

Reviewer: ```suggestion
If everything passes the checks, the assets move from the `payment` to
```

After a successful deposit, ERTP guarantees:
- The `payment` is burned (i.e. destroyed).
- The `purse` contains the total of what it held before plus the `payment`'s full content.
  - i.e. If the `purse` had 7 Quatloos and the `payment` had 3 Quatloos, after depositing the `payment`
    the `purse` has 10 Quatloos.

When the `deposit()` call throws an error (i.e. something went wrong),
ERTP guarantees: 
- The alleged `payment` is in the same state as before the call.
- The `purse` is in the same state as before the call.

In other words, a failed attempt to deposit a 3 Quatloo `payment`
in a 7 Quatloo `purse` means the `payment` continues to exist and hold
3 Quatloos, and the `purse` continues to hold 7 Quatloos.

## Issuers and mints

Other key ERTP components are:

- **Mints**: Make new digital assets as a new `Payment`. `Mints` only
make one kind of asset (Quatloos, Simoleons, Moola, other imaginary
currencies, etc.). We refer to
that kind as a `mint`'s *Brand*. So if a `mint` issues Quatloos, it's a
Quatloo `brand` `mint`.  Only `mints` can issue new digital assets. To mint
new assets of a particular kind, you must have a reference to that

> kind's `mint` 

Reviewer: ```suggestion
kind's `mint`. 
```

> - **Issuers**: Create empty `purses`, manipulate and operate on
>  `payments` and map minted assets to both of them when assets are 
>  added or removed. `Issuers` verify and move digital assets.

Reviewer: This is a lot of internal detail in an intro, I think. How about:

```suggestion
- **Issuers**: Create empty `purses`, and manipulate and operate on
 `payments`. `Issuers` are the authority on which payments and purses hold what digital assets.
```

An `issuer`'s special admin facet is a `mint`, and that `mint` and `issuer`
have a one-to-one relationship. With a reference to an `issuer`, you can

> check the validity of a `payment` in that `issuer`'s `mint`'s assets, as

Reviewer: 'Issuer's mint's assets" doesn't make sense. How about instead:
```suggestion
check the validity of a `payment`, as
```

well as claim it either as a new `payment` to yourself or a `purse` you
control. 

> `Issuers` (i.e. their references) should be gotten from a trusted source

Reviewer: ```suggestion
`Issuers` should be gotten from a trusted source
```
I didn't understand `(i.e. their references)`

and then relied upon as the decider of whether an untrusted `payment` is

> valid  

Reviewer: ```suggestion
valid.  
```

> **Note**: There is a one-to-one correspondence between a `brand` and a
>   `mint`, a `mint` and an `issuer`, and an `issuer` and a `brand`. In other

Reviewer: ```suggestion
**Note**: There is a one-to-one correspondence between a `brand`, 
  `mint`, and `issuer`. In other
```

their creation. So a `purse` might hold Quatloos, meaning it couldn't
hold Moola or any other non-Quatloo asset). Note: Quatloos and Moola are both
imaginary currencies.

If everything passes the checks, the assets move from the `payment` to
the `purse`. If there's a problem, it throws an error.

After a successful deposit, ERTP guarantees:
- The `payment` is burned (i.e. destroyed).
- The `purse` contains the total of what it held before plus the `payment`'s full content.
  - i.e. If the `purse` had 7 Quatloos and the `payment` had 3 Quatloos, after depositing the `payment`
    the `purse` has 10 Quatloos.

When the `deposit()` call throws an error (i.e. something went wrong),
ERTP guarantees: 
- The alleged `payment` is in the same state as before the call.
- The `purse` is in the same state as before the call.

In other words, a failed attempt to deposit a 3 Quatloo `payment`
in a 7 Quatloo `purse` means the `payment` continues to exist and hold
3 Quatloos, and the `purse` continues to hold 7 Quatloos.

## Issuers and mints

Other key ERTP components are:

- **Mints**: Make new digital assets as a new `Payment`. `Mints` only

> make one kind of asset (Quatloos, Simoleons, Moola, other imaginary
> currencies, etc.). We refer to

Reviewer 2: We shouldn't imply that we only work with imaginary currencies!

```suggestion
make one kind of asset (we use fictional currencies in our examples, but they can represent actual currencies, in-game assets, real property, etc.). We refer to
```

kind's `mint`. 

- **Issuers**: Create empty `purses`, and manipulate and operate on
 `payments`. `Issuers` are the authority on which `payments` and `purses`
 hold what digital assets.

An `issuer`'s special admin facet is a `mint`, and that `mint` and `issuer`
have a one-to-one relationship. With a reference to an `issuer`, you can
check the validity of a `payment` whose asset `brand` is associated with that `issuer`;
i.e. If you have a reference to the Quatloos `issuer`, you can validate
any `payment` made in Quatloos. You can also claim the `payment` either 
as a new `payment` to yourself or a `purse` you control. 

`Issuers` should be gotten from a trusted source
and then relied upon as the decider of whether an untrusted `payment` is valid. 

**Note**: There is a one-to-one correspondence between a `brand`, a
  `mint`, and an `issuer`. In other words:
  - A `mint` associated with a Quatloos `brand` can only create new Quatloos
  and is the only `mint` that can create new Quatloos. 
  - A `issuer` associated with a Quatloos `mint` can only operate on Quatloos
  asset holders. It is the only `issuer` that can operate on them. 

## Amounts

> *Amounts* describe digital assets without having any value of their own.
> Anyone can make one, and they can be sent to anyone. They have two parts:

Reviewer 2: ```suggestion
*Amounts* describe digital assets without having any value of their own.
Anyone can make one, and they can be freely sent to anyone since they
convey no underlying value. They have two parts:
```

second balcony far left  (and affects what you're willing to trade for
it). 

## AmountMath

`Issuers` must be able to deposit and withdraw assets from a `purse`. This
requires being able to add and subtract digital assets. They use a set
of `amountMath` functions.

In addition to math operations, `amountMath` functions check on their
arguments' `brands`, throwing an error if the wrong `brand` was used.

> `amountMaths` only work on assets of their associated `brand` and its `issuer.

Reviewer: ```suggestion
`amountMath` only work on assets of their associated `brand` and `issuer`.
```

There can be many copies of the `amountMath` for a particular `brand` and
its `issuer`. 

## Next Steps

> If you are Getting Started, you should go to the [Introduction to Zoe](https://agoric.com/documentation/getting-started/intro-zoe.html).

Reviewer: Can we please use relative links so that the version running locally also links locally?
```suggestion
If you are Getting Started, you should go to the <router-link to="/getting-started/intro-zoe.html">Introduction to Zoe</router-link>.
```

If you've finished the Getting Started material, you should go to the

> [ERTP Guide](https://agoric.com/documentation/ertp/guide/) for a fuller explanation of ERTP

Reviewer: ```suggestion
<router-link to="/ertp/guide/">ERTP Guide</router-link> for a fuller explanation of ERTP
```
